plugins/push/frontend/public/javascripts/countly.models.js

```diff
@@ -99,271 +214,718 @@
-        },
-        getLocalizationFilterOptions: function() {
-            var self = this;
-            return [
-                {label: CV.i18n("push-notification-details.localization-filter-all"), value: self.LocalizationEnum.ALL}
-            ];
+        getDefaultLocalization: function() {
+            return {label: DEFAULT_LOCALIZATION_LABEL, value: DEFAULT_LOCALIZATION_VALUE, percentage: 100};
         },
-        getTypeUrlParameter: function(type) {
-            if (type === this.TypeEnum.AUTOMATIC) {
-                return {auto: true, tx: false};
+        hasNoUsersToSendPushNotification: function(pushNotificationDto) {
+            return pushNotificationDto.build.total === 0;
+        },
+        isNoPushCredentialsError: function(error) {
+            return error.message === 'No push credentials';
+        },
+        isNoUsersFoundError: function(error) {
+            return error.message === 'No users were found from selected configuration';
+        },
+        convertDateTimeToMS: function(datetime) {
+            var result = 0;
+            if (datetime.days) {
+                result += moment.duration(parseInt(datetime.days, 10), 'd').asMilliseconds();
             }
-            if (type === this.TypeEnum.TRANSACTIONAL) {
-                return {auto: false, tx: true};
+            if (datetime.hours) {
+                result += moment.duration(parseInt(datetime.hours, 10), 'h').asMilliseconds();
             }
-            return { auto: false, tx: false};
+            return result;
         },
-        mapType: function(dto) {
-            if (dto.tx === false && dto.auto === true) {
-                return this.TypeEnum.AUTOMATIC;
+        convertMSToDaysAndHours: function(dateTimeInMs) {
+            var days = parseInt(dateTimeInMs / DAY_TO_MS_RATIO, 10);
+            var remainingTime = (dateTimeInMs % DAY_TO_MS_RATIO);
+            var hours = parseInt(remainingTime / HOUR_TO_MS_RATIO, 10);
+            return {
+                days: days,
+                hours: hours
+            };
+        },
+        formatDateTime: function(dateTime, format) {
+            if (dateTime === 'Never') {
+                return dateTime;
             }
-            if (dto.tx === true && dto.auto === false) {
-                return this.TypeEnum.TRANSACTIONAL;
+            if (!format) {
+                format = "DD.MM.YYYY hh:mm";
             }
-            return this.TypeEnum.ONE_TIME;
+            return moment(dateTime).format(format);
         },
-        mapStatus: function(status, error) {
-            if (StatusFinderHelper.isSending(status)) {
-                if (error) {
-                    return {value: 'sending-errors', label: CV.i18n('push-notification.status-sending-errors')};
+    };
+
+    countlyPushNotification.mapper = {
+        incoming: {
+            getUserPropertyElement: function(index, userProperty) {
+                var newElement = document.createElement("span");
```

Comment:
I cannot get why createElement is needed here (similar to latter occurrences). It looks a bit hacky I can tell you that, but more importantly, It is placed in models.js

Reply:
If it is related to string unescaping as in one of your other tickets, I don't think that it is a good practice. Because it allows all types of unescaping, which might be dangerous.

Reply:
I should have added a comment perhaps explaining the purpose of the mappers as well as helper methods. This is an incoming mapper (it maps the server response). The reason is because the message title and content needs to be constructed given the user properties that it might have, e.g. at index position 5, insert a specific user property in message. I chose to do it in the model so that I don't have to construct the message everytime I use it. The incoming escaped content is simply processed.